From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Thu, 20 Jul 2023 19:13:36 +0200
Subject: [PATCH] qt: Unbreak build with qt-egl enabled but viv_fb missing

Avoid an error message when the feature is explicitly enabled:

    ERROR: Feature qt-egl cannot be enabled: gstreamer-gl-viv_fb-1.0 is required
---
 subprojects/gst-plugins-good/ext/qt/meson.build | 17 ++++++-----------
 1 file changed, 6 insertions(+), 11 deletions(-)

diff --git a/subprojects/gst-plugins-good/ext/qt/meson.build b/subprojects/gst-plugins-good/ext/qt/meson.build
index 442b7d5870e1..d1db2514653e 100644
--- a/subprojects/gst-plugins-good/ext/qt/meson.build
+++ b/subprojects/gst-plugins-good/ext/qt/meson.build
@@ -120,6 +120,12 @@ if qt5_egl.allowed()
   qt_defines += ['-DHAVE_QT_EGLFS']
   optional_deps += gstglegl_dep
   have_qt_windowing = true
+
+  # EGL windowing for Vivante Framebuffer (e.g. i.MX6)
+  if gstglviv_fb_dep.found()
+    qt_defines += ['-DHAVE_QT_VIV_FB']
+    optional_deps += gstglviv_fb_dep
+  endif
 endif
 
 # Android windowing
@@ -184,17 +190,6 @@ if host_system == 'ios'
   endif
 endif
 
-# EGL windowing for Vivante Framebuffer (e.g. i.MX6)
-qt5_viv_fb = qt5_egl \
-    .require(host_system == 'linux') \
-    .require(gstglviv_fb_dep.found(), error_message: 'gstreamer-gl-viv_fb-1.0 is required') \
-    .require(gst_gl_have_platform_egl, error_message: 'egl platform support in gstreamer-gl is required')
-if qt5_viv_fb.allowed()
-  qt_defines += ['-DHAVE_QT_VIV_FB']
-  optional_deps += gstglviv_fb_dep
-  have_qt_windowing = true
-endif
-
 if qt5_option.require(have_qt_windowing).allowed()
   # rpath is needed to be able to load the plugin on macOS inside the devenv
   qmlgl_kwargs = {}
